usv_gazebo_plugins/src/usv_gazebo_dynamics_plugin.cc

```diff
@@ -328,7 +328,9 @@ void UsvDynamicsPlugin::Update()
       double simTime = kTimeNow.Double();
       // double depth = WavefieldSampler::ComputeDepthDirectly(
       //  *waveParams, X, simTime);
-      double depth = WavefieldSampler::ComputeDepthSimply(
+      double depth = 0.0;
+      if (waveParams)
```

Comment:
Does the check on https://github.com/osrf/vrx/blob/master/usv_gazebo_plugins/src/usv_gazebo_dynamics_plugin.cc#L182-L188 not give it a non-null value?

Reply:
In one of my use cases, it does not since that world does not have a wave model.

While developing the USV model, I often load it in an empty gazebo world, which worked fine until I added the dynamics plugin which led to a crash without many clues to what happened.

Now I understand that the plugin is meant to be used in a world with waves, but I think that gazebo shouldn't crash in the case the wave model is missing. Perhaps a warning or error message might be better, but this is just a quick fix to prevent the crash.

Reply:
Yeah, I think it is meant to be loaded with waves. I also see your point that the code should be more robust and shouldn't crash on a null pointer.
Just curious, do you get that message on line 184 printed over and over again?

Reply:
No, I don't get that message at all. I tried with and without the debug flag on when launching gazebo.

Reply:
Update: I just found the verbose flag and that message does continuously print with it on.

Reply:
Oh yeah that's right, it won't show up in normal mode.